src/python/zquantum/core/testing/generate_cases_for_backend_tests.py

```diff
@@ -0,0 +1,333 @@
+            [1j * sympy.sin(theta), 0, 0, sympy.cos(theta)],
+        ]
+    )
+    YY = sympy.Matrix(
+        [
+            [sympy.cos(theta), 0, 0, 1j * sympy.sin(theta)],
+            [0, sympy.cos(theta), -1j * sympy.sin(theta), 0],
+            [0, -1j * sympy.sin(theta), sympy.cos(theta), 0],
+            [1j * sympy.sin(theta), 0, 0, sympy.cos(theta)],
+        ]
+    )
+    ZZ = sympy.Matrix(
+        [
+            [sympy.cos(theta) + 1j * sympy.sin(theta), 0, 0, 0],
+            [0, sympy.cos(theta) - 1j * sympy.sin(theta), 0, 0],
+            [0, 0, sympy.cos(theta) - 1j * sympy.sin(theta), 0],
+            [0, 0, 0, sympy.cos(theta) + 1j * sympy.sin(theta)],
+        ]
+    )
+
+    angles = [-sympy.pi / 2, 0, sympy.pi / 5, sympy.pi / 2, sympy.pi]
+    print("**" * 10)
+    print("WAVEFUNCTION")
+    print("-" * 10)
+    print("1 qubit gates")
+    print("-" * 10)
+    generate_cases_1_qubit_wavefunction(Rx, "Rx", angles)
+    generate_cases_1_qubit_wavefunction(Ry, "Ry", angles)
+    generate_cases_1_qubit_wavefunction(Rz, "Rz", angles)
+    generate_cases_1_qubit_wavefunction(PHASE, "PHASE", angles)
+    generate_cases_1_qubit_wavefunction(RH, "RH", angles)
+    print("-" * 10)
+    print("2 qubit gates")
+    print("-" * 10)
+    generate_cases_2_qubits_wavefunction(CPHASE, "CPHASE", angles)
+    generate_cases_2_qubits_wavefunction(XX, "XX", angles)
+    generate_cases_2_qubits_wavefunction(YY, "YY", angles)
+    generate_cases_2_qubits_wavefunction(ZZ, "ZZ", angles)
+
+    print("**" * 10)
+    print("EXP VALS WITHOUT ANGLES")
+    print("-" * 10)
+    print("1 qubit gates")
+    print("-" * 10)
+    generate_cases_1_qubit_exp_vals(X, "X")
+    generate_cases_1_qubit_exp_vals(Y, "Y")
+    generate_cases_1_qubit_exp_vals(Z, "Z")
+    generate_cases_1_qubit_exp_vals(H, "H")
+    generate_cases_1_qubit_exp_vals(S, "S")
+    generate_cases_1_qubit_exp_vals(T, "T")
+    print("-" * 10)
+    print("2 qubit gates")
+    print("-" * 10)
+    generate_cases_2_qubits_exp_vals(CNOT, "CNOT")
+    generate_cases_2_qubits_exp_vals(SWAP, "SWAP")
+    generate_cases_2_qubits_exp_vals(CZ, "CZ")
+
+    print("**" * 10)
+    print("EXP VALS WITH ANGLES")
+    print("-" * 10)
+    print("1 qubit gates")
+    print("-" * 10)
+    generate_cases_1_qubit_exp_vals_with_angles(Rx, "Rx", angles)
+    generate_cases_1_qubit_exp_vals_with_angles(Ry, "Ry", angles)
+    generate_cases_1_qubit_exp_vals_with_angles(Rz, "Rz", angles)
+    generate_cases_1_qubit_exp_vals_with_angles(PHASE, "PHASE", angles)
+    generate_cases_1_qubit_exp_vals_with_angles(RH, "RH", angles)
+    print("-" * 10)
+    print("2 qubit gates")
+    print("-" * 10)
+    generate_cases_2_qubits_exp_vals_with_angles(CPHASE, "CPHASE", angles)
+    generate_cases_2_qubits_exp_vals_with_angles(XX, "XX", angles)
+    generate_cases_2_qubits_exp_vals_with_angles(YY, "YY", angles)
+    generate_cases_2_qubits_exp_vals_with_angles(ZZ, "ZZ", angles)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
This seems like something that isn't supposed to be here anymore :)

Reply:
It actually should – these functions are not executed anywhere outside of this script, so leaving it here makes it easy to use and generate new cases :)

Reply:
Gotcha, this confusion goes along with this comment: https://github.com/zapatacomputing/z-quantum-core/pull/145#discussion_r554165870

Maybe a quick comment saying something like: `This script is used to generate test cases in the file <insert-filename>. Run with python3 <script>.py`

Reply:
Addressed in [46e52f](https://github.com/zapatacomputing/z-quantum-core/pull/145/commits/46e52f81e7ca383dc39a0b581a221a5a3c2693c3).